pkg/sql/pg_metadata_diff.go

```diff
@@ -86,8 +85,14 @@ type PGMetadataColumnType struct {
 // PGMetadataColumns maps columns names to datatypes.
 type PGMetadataColumns map[string]*PGMetadataColumnType
 
+// PGMetadataTableInfo represents a table with column mapping and insertion order.
+type PGMetadataTableInfo struct {
+	Order   *[]string         `json:"order"`
```

Comment:
nit: why is this a pointer to a slice? 

Another nit, can we rename order to `ColumnNames` and leave a comment that it must be in sorted order.